mpags-cipher.cpp

```diff
@@ -0,0 +1,244 @@
+
+            helpRequested = true;
+
+        }
+
+        // Looking for whether the version is requested
+        if(cmdLineArgs[i] == "--version" ){
+
+            versionRequested = true;
+
+        }
+
+
+        // Looking for whether the option -i has been used as an argument
+        if(cmdLineArgs[i] == "-i" ){
+            
+            // Need to look to see if this was the last argument specified
+            // If it was, need to return an error message.
+            if (i == (cmdLineArgs.size() - 1)){
+                std::cout << "An input file must be specified in the -i option after '-i'." << std::endl;
+                break;
+            }
+            else{
+                // If it wasn't, proceed as normal.
+
+                // Take the next argument as the input file
+                input_fileGiven = true;
+                //std::cout << input_fileGiven << std::endl;
+                input_file = cmdLineArgs[i+1];
+                //std::cout << "The file identified as an input file is " << input_file << std::endl;
+
+            }
+
+            
+        }
+        
+
+
+
+        // Looking for whether the option -o has been used as an argument
+        if(cmdLineArgs[i] == "-o" ){
+
+            // Need to look to see if this was the last argument specified
+            // If it was, need to return an error message.
+            if (i == (cmdLineArgs.size() - 1)){
+                std::cout << "An output file must be specified in the -o option after '-o'." << std::endl;
+                break;
+            }
+            else{
+                // If it wasn't, proceed as normal.
+            
+                // Take the next argument as the output file
+                output_fileGiven = true;
+                output_file = cmdLineArgs[i+1];
+                //std::cout << "The file identified as an output file is " << output_file << std::endl;
+            }
+           
+        }
+        
+
+        
+        
+    }
+
+
+    
+
+    // When help is requested
+    if (helpRequested) {
+        // Print help message
+        std::cout
+        << " \n"
+        << " mpags-cipher [-h/--help]\n" 
+        << " Encrypt and decrypt alphanumeric text using classical ciphers\n"
+        << " Avaliable options:\n"
+        << " -h | --help            Print this help message and exit\n"
+        << " --version              Print the version number and exit\n"
+        << " -i <input_file>        Specify the input file\n"
+        << " -o <output_file>       Specify the output file\n"
+        << " -c <cipher>            Specify the type of classical cipher:\n"
+        << "                                type1\n"
+        << "                                type2\n"
+        << "                                type3\n"
+        << " -k <key>               Specify the key\n"
+        << " encrypt/decrypt        Specify if the message is for encryption or decryption\n"
```

Comment:
Good that you're thinking ahead but at the moment the program doesn't support these options, so they probably shouldn't be included in the help printout until they are supported by the code.